scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder().apply {
+        url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      }.build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+    }.build()
+    val updatedLicense2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+      this.extractedCopyLink = ExtractedCopyLink.newBuilder().apply {
+        url = "https://local-copy/bsd-license"
+      }.build()
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val finalDepsList = MavenDependenciesRetriever.updateMavenDependenciesList(
+      mavenDependenciesList,
+      setOf<License>(updatedLicense2)
+    )
+    assertThat(finalDepsList.size).isEqualTo(2)
+    assertIsDependency(
+      dependency = finalDepsList[0],
+      artifactName = DEP_WITH_SCRAPABLE_LICENSE,
+      artifactVersion = DATA_BINDING_VERSION
+    )
+    verifyLicenseHasScrapableVerifiedLink(
+      license = finalDepsList[0].licenseList[0],
+      licenseName = "The Apache License, Version 2.0",
+      originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt",
+      verifiedLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    )
+    assertIsDependency(
+      dependency = finalDepsList[1],
+      artifactName = DEP_WITH_NO_LICENSE,
+      artifactVersion = PROTO_LITE_VERSION
+    )
+    verifyLicenseHasExtractedCopyVerifiedLink(
+      license = finalDepsList[1].licenseList[0],
+      licenseName = "Simplified BSD License",
+      originalLink = "https://www.opensource.org/licenses/bsd-license",
+      verifiedLink = "https://local-copy/bsd-license"
+    )
+  }
+
+  @Test
+  fun testWriteTextProto_emptyMavenDependencyList_textProtoFileEmpty() {
+    val textProtoFile = tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    val mavenDependencyList = MavenDependencyList.newBuilder().build()
+
+    MavenDependenciesRetriever.writeTextProto(
+      "${tempFolder.root}/scripts/assets/maven_dependencies.textproto",
+      mavenDependencyList
+    )
+
+    assertThat(textProtoFile.readAsJoinedString()).isEqualTo("")
```

Comment:
Could instead assert that the string is empty.

Reply:
Done.